src/app/modules/submission/submission.utils.ts

```diff
@@ -68,3 +92,253 @@ export const extractEmailConfirmationData = (
     return acc
   }, [])
 }
+
+/**
+ * Extracts response data to be sent in email confirmations
+ * @param responses Responses from form filler
+ * @param formFields Fields from form object
+ * @returns Array of data for email confirmations
+ */
+export const extractEmailConfirmationDataFromIncomingSubmission = (
```

Comment:
any reason why this function is not a class method of `IncomingSubmission`? Separate responsibilities?

`incomingSubmission.extractEmailConfirmationData` seems to make sense

Reply:
^ I had the exact same thought

Reply:
you're right, my intent was to minimise the amount of logic related to the representation of data within this data class itself.

Eventually an `IncomingSubmission` will need to emit various projections of its internal data for different purposes. Some would be for the data collation tool etc, and others like this for extracting email data/recipients. Having all of these different representations baked into class methods will work, albeit messily especially as we scale and increase the feature set (like sgid).

the idea i was thinking of was incorporating aspects of the strategy pattern (like in sendEmailConfirmation, which uses either this function or the old`extractEmailConfirmationData` depending on the data structure), and double-dispatch (which can hopefully help to cleanly project data into `SubmissionEmailObject`).

will definitely need your input whether this is a premature optimisation or sth worth keeping/working towards